scrapy/settings/__init__.py

```diff
@@ -88,19 +106,30 @@ def getdict(self, name, default=None):
             value = json.loads(value)
         return dict(value)
 
+    def getpriority(self, name):
+        prio = None
+        if name in self.attributes:
+            prio = self.attributes[name].priority
+        return prio
+
+    def __setitem__(self, name, value):
+        self.set(name, value)
+
     def set(self, name, value, priority='project'):
         self._assert_mutability()
-        if isinstance(priority, six.string_types):
-            priority = SETTINGS_PRIORITIES[priority]
+        priority = get_settings_priority(priority)
         if name not in self.attributes:
-            self.attributes[name] = SettingsAttribute(value, priority)
+            if isinstance(value, SettingsAttribute):
+                self.attributes[name] = value
+            else:
+                if isinstance(value, dict):
+                    value = BaseSettings(value, priority)
+                self.attributes[name] = SettingsAttribute(value, priority)
```

Comment:
I agree. Now that we split `Settings` and `BaseSettings` explicit promotion of only the default settings seems a more sane choice.